File: go/day82.go
package main

import (
	"fmt"
	"bufio"
	"os"
	"strings"
	"regexp"
)
func main() {
	fmt.Println("please input card number");


	reader := bufio.NewReader(os.Stdin)


	cardNumber, _ := reader.ReadString('\n');

	cardNumber = strings.TrimSpace(cardNumber);


	match, _ := regexp.MatchString("^[\\d ]{13,19}$", cardNumber)

	if match {
		fmt.Println("Card number is valid.")
	} else {
		fmt.Println("Card number is invalid!!!!")
	}
}

File: cSharp/day82.cs
using System;
using System.Text.RegularExpressions;

Console.WriteLine("Please input your credit/debit card number:");

string cardNumber1 = Console.ReadLine();

Regex cardNumberPattern = new Regex("^[\\d ]{13,19}$");

if (cardNumberPattern.Match(cardNumber1).Success) {
    Console.WriteLine("Card number is valid.");
}
else {
    Console.WriteLine("Card number is invalid !!!!");
}